src/platform/Darwin/DnssdContexts.cpp

```diff
@@ -156,6 +156,9 @@ CHIP_ERROR MdnsContexts::Add(GenericContext * context, DNSServiceRef sdRef)
     auto err = DNSServiceSetDispatchQueue(sdRef, chip::DeviceLayer::PlatformMgrImpl().GetWorkQueue());
     if (kDNSServiceErr_NoError != err)
     {
+        // We can't just use our Delete to deallocate the service ref here,
+        // because our context may not have its serviceRef set yet.
+        DNSServiceRefDeallocate(sdRef);
```

Comment:
Is this still needed ?

Reply:
I think it's needed in general, yes.  When we get here, we have already allocated the service ref, and if we don't clean it up here nothing will ever clean it up....  So it's not strictly related to the other changes here, just something I found while working on this.

Reply:
Doing a quick look reveal that you are correct. I wonder if we should clear the context too at https://github.com/project-chip/connectedhomeip/pull/24733/files/e41aedc6fa18fa4fcba4c67734e2a5429c05e61e#diff-d3d1d99d691d0aef90e9de721cad391450d0e28aa0b727d812022acc3755f248R148

Reply:
So up above where we check for `context == nullptr`?  Yes, we should, if sdRef is not null...  Will do that.